pkg/api/unversioned/group_version.go

```diff
@@ -198,3 +198,46 @@ func (gv *GroupVersion) UnmarshalJSON(value []byte) error {
 func (gv *GroupVersion) UnmarshalText(value []byte) error {
 	return gv.unmarshal(value)
 }
+
+// ToAPIVersionAndKind is a convenience method for satisfying runtime.Object on types that
+// do not use TypeMeta.
+func (gvk *GroupVersionKind) ToAPIVersionAndKind() (string, string) {
+	if gvk == nil {
+		return "", ""
+	}
+	return gvk.GroupVersion().String(), gvk.Kind
+}
+
+// FromAPIVersionAndKind returns a GVK representing the provided fields for types that
+// do not use TypeMeta.
+func FromAPIVersionAndKind(apiVersion, kind string) *GroupVersionKind {
```

Comment:
I'm less than thrilled with this.  I was hoping to avoid having enough string representations of `apiVersion` and `kind` to ever want this method.  Can this be made local to its point of use?  Or maybe take a `TypeMeta`?